From da754a62c0f642bf1d13799fdf577323753633c4 Mon Sep 17 00:00:00 2001
From: Franck Lenormand <franckx.lenormand@intel.com>
Date: Mon, 18 May 2015 17:38:55 +0200
Subject: [PATCH] [BT] System hang and crash on BT ON/OFF

When the uart driver for the BT is enable
or disable, it called pm_runtime_idle with
a bad context

Tracked-On: https://jira01.devtools.intel.com/browse/GMINL-9549
Change-Id: Ib801738ff172c9ff93d7c5a32724df8a71afedfa
Signed-off-by: franck Lenormand <franckx.lenormand@intel.com>
---
 drivers/misc/bt_lpm/bt_lpm.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/misc/bt_lpm/bt_lpm.c b/drivers/misc/bt_lpm/bt_lpm.c
index 688c4cddf704..6cddaae29d13 100644
--- a/drivers/misc/bt_lpm/bt_lpm.c
+++ b/drivers/misc/bt_lpm/bt_lpm.c
@@ -79,14 +79,14 @@ static void uart_enable(struct device *tty)
 {
 	pr_debug("%s: runtime get\n", __func__);
 	/* Tell PM runtime to power on the tty device and block s0i3 */
-	pm_runtime_get_sync(tty);
+	pm_runtime_get(tty);
 }
 
 static void uart_disable(struct device *tty)
 {
 	pr_debug("%s: runtime put\n", __func__);
 	/* Tell PM runtime to release tty device and allow s0i3 */
-	pm_runtime_put_sync(tty);
+	pm_runtime_put(tty);
 }
 #endif /* !DBG_DISABLE_BT_LOW_POWER */
 
-- 
1.9.1

